File: python/409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        answer = 0
        freq = [0] * 128
        for i in s:
            freq[ord(i)] += 1
        odd = False
        for i in freq:
            if i % 2 == 0: answer += i
            else:
                if odd:
                    answer += i-1
                else:
                    answer += i
                    odd = True
        return answer

File: java/409. Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] freq = new int[128];
        for (int i = 0; i < s.length(); i++) {
            freq[s.charAt(i)]++;
        }
        int answer = 0;
        boolean odd = false;
        for (int i = 0; i < freq.length; i++) {
            if (freq[i] % 2 == 0) answer += freq[i];
            else {
                if (odd) {
                    answer += freq[i] - 1;
                } else {
                    answer += freq[i];
                    odd = true;
                }
            }
        }
        return answer;
    }
}